content/advent-2018/go-arrow.md

```diff
@@ -0,0 +1,531 @@
+- [FITS](https://godoc.org/github.com/astrogo/fitsio)
+- [HDFS](https://godoc.org/github.com/colinmarc/hdfs)
+- [Parquet](https://godoc.org/github.com/xitongsys/parquet-go/parquet)
+- [ORC](https://godoc.org/github.com/scritchley/orc)
+- [Cassandra](https://godoc.org/github.com/gocql/gocql)
+- ...
+
+Many basic ingredients are already there, even if some are still missing.
+
+But even if basic file-level interoperability is (somewhat) achieved, one still needs to implement readers, writers and converters from one file format to any another when integrating with a new analysis pipeline:
+
+![file formats](/postimages/advent-2018/go-arrow/arrow-copy.png)
+
+In many scenarios, this implies a large fraction of the computation is wasted on serialization and deserialization, reimplementing over and over the same features for converting from `format-1` to `format-n` or from `format-1` to `format-m`, _etc..._
+
+What if we had a common data layer between all these analysis pipelines, a common tool that would also be efficient?
+
+## Arrow
+
+What is [Arrow](https://arrow.apache.org)? From the website:
+
+```
+Apache Arrow is a cross-language development platform for in-memory data.
+
+It specifies a standardized language-independent columnar memory format for flat
+and hierarchical data, organized for efficient analytic operations on modern
+hardware.
+It also provides computational libraries and zero-copy streaming messaging and
+interprocess communication.
+```
+
+![arrow shared](/postimages/advent-2018/go-arrow/arrow-shared.png)
+
+The idea behind Arrow is to describe and implement a cross-language mechanism for efficiently sharing data across languages and processes.
+This is embodied as an `arrow::array` in C++, the reference implementation.
+Languages currently supported include C, C++, Java, JavaScript, Python, and Ruby.
+
+... and -- of course -- Go :)
+
+[InfluxData](https://www.influxdata.com/) contributed the original Go code for Apache Arrow, as announced [here](https://www.influxdata.com/blog/influxdata-apache-arrow-go-implementation/).
+Initially, the Go Arrow package implemented by [Stuart Carnie](https://twitter.com/stuartcarnie) had support for:
+
+- primitive arrays (`{u,}int{8,16,32,64}`, `float{32,64}`, ...)
+- parametric types (`timestamp`)
+- memory management
+- typed metadata
+- SIMD math kernels (_via_ a nice automatic translation tool - `c2goasm` - to extract vectorized code produced by CLang, see the [original blog post](https://www.influxdata.com/blog/influxdata-apache-arrow-go-implementation/) for more details)
+
+Since then, a few contributors provided implementations for:
+
+- list arrays
+- struct arrays
+- time arrays
+- loading `CSV` data to Arrow arrays,
+- tables and records,
+- tensors (_a.k.a._ n-dimensional arrays.)
+
+But what are those arrays?
+
+## Arrow arrays
+
+From the Arrow perspective, an array is a sequence of values with known length, all having the same type.
```

Comment:
You talk about nested data below but since this is a key feature of the columnar format, it might be worth mentioning earlier that efficient representation of nested / semi-structured data is a key feature of the project, and one that many tabular data projects don't do well

Reply:
done (I think)